test/LiveServer.hs

```diff
@@ -321,6 +321,24 @@ liveServerUnitTests = testGroup "Unit tests against a live server"
       uploads <- listIncompleteUploads bucket (Just object) False C.$$ sinkList
       liftIO $ (null uploads) @? "removeIncompleteUploads didn't complete successfully"
 
+  , funTestWithBucket "putObject contentType tests" $ \step bucket -> do
+      step "fPutObject content type test"
+      let object = "xxx-content-type"
+          size1 = 100 :: Int64
+
+      step "create server object with content-type"
+      inputFile <- mkRandFile size1
+      fPutObject bucket object inputFile def{
+        pooContentType = Just "application/javascript"
+        }
+
+      -- retrieve obj info to check
+      ObjectInfo _ _ _ _ m <- headObject bucket object
```

Comment:
To make tests backward compatible, its best to avoid pattern matching on unexported constructor. You could instead,
```hs
oi <- headObject bucket object
let m = oiMetadata oi
```

Reply:
Thanks for catching that..had missed it.